### PR TITLE
Delete Users use case

### PR DESCRIPTION
**Pull request by wkukka1**

### CODE REVIEW

File: src/main/java/interface_adapter/delete_user/DeleteViewModel.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package interface_adapter.delete_user;
+
+import interface_adapter.ViewModel;
+
+import java.beans.PropertyChangeListener;
+import java.beans.PropertyChangeSupport;
+
+public class DeleteViewModel extends ViewModel {
+    private DeleteState state = new DeleteState();
```
Comment (K-Logo) [inline src/main/java/interface_adapter/delete_user/DeleteViewModel.java:9]: Don't think you need this

**Review by K-Logo — DISMISSED** [review]

File: src/main/java/interface_adapter/delete_user/DeleteState.java
[the remark this answers is not in the document]
Comment (wkukka1) [reply src/main/java/interface_adapter/delete_user/DeleteState.java:1]: I need the DeleteState to be able to use the property fire change observer

File: src/main/java/interface_adapter/delete_user/DeleteViewModel.java (hunk 1 above)
Comment (wkukka1) [reply src/main/java/interface_adapter/delete_user/DeleteViewModel.java:9]: I need the DeleteState to be able to use the property fire change observer

**Review by K-Logo — APPROVED** [review]
Good naming conventions (camelCase). Although I got a little confused when you named your DeleteInputBoundary object deleteInteractor. 
Handy inclusion of the test file (it does run and passes the test). 
Implements the Clean Architecture engine very well. The code also conforms to the SOLID principles.
One suggestion I have is to include documentation (Javadoc) on some of the methods, just in case, and for future implementations. 
Good work!

**Pull request merged**